crypto/params.c

```diff
@@ -62,26 +63,173 @@ void OSSL_PARAM_set_all_unmodified(OSSL_PARAM *p)
             p++->return_size = OSSL_PARAM_UNMODIFIED;
 }
 
+/* Return non-zero if the signed number is negative */
+static int is_negative(const void *number, size_t s)
+{
+    const unsigned char *n = number;
+    DECLARE_IS_ENDIAN;
+
+    return 0x80 & (IS_BIG_ENDIAN ? *n : n[s - 1]);
+}
+
+/* Check that all the bytes specified match the expected sign byte */
+static int check_sign_bytes(const unsigned char *p, size_t n, unsigned char s)
+{
+    size_t i;
+
+    for (i = 0; i < n; i++)
+        if (p[i] != s)
+            return 0;
+    return 1;
+}
+
+/*
+ * Copy an integer to another integer.
+ * Handle different length integers and signed and unsigned integers.
+ * Both integers are in native byte ordering.
+ */
+static int copy_integer(unsigned char *dest, size_t dest_len,
+                        const unsigned char *src, size_t src_len,
+                        unsigned char pad, int signed_int)
+{
+    size_t n;
+    DECLARE_IS_ENDIAN;
+
+    if (IS_BIG_ENDIAN) {
+        if (src_len < dest_len) {
+            n = dest_len - src_len;
+            memset(dest, pad, n);
+            memcpy(dest + n, src, src_len);
+        } else {
+            n = src_len - dest_len;
+            if (!check_sign_bytes(src, n, pad)
+                    || (signed_int && ((pad ^ src[n]) & 0x80) != 0))
```

Comment:
I'm not sure I understand what this expression does...  would you mind explaining it?

Reply:
Oh never mind, now I get it, I didn't notice that you implemented signed\<-\>unsigned coersion!

Reply:
I'll add a comment, this actually has nothing to do with signed<->unsigned conversions.

E.g. when converting the sixteen bit signed integer -253 to an eight bit signed value: FF03 truncates to 03 and passes the `check_sign_bytes` condition.